### ISSUE TITLE
.ONNX -> .PLAN successful conversion but the generated .plan file is small

### ISSUE THREAD

**Issue by muzhibo1314** [NVIDIA/TensorRT#1331]
## Description

When I was converting from .onnx to .plan format, an error was reported that the onehot plugin does not support it, so I wrote a onehot custom plugin, but I don’t know about custom plugins. This problem occurred after I finished writing. Does anyone know?  thanks

<img width="416" alt="WeChat3abe915a87f4f028ce59e455df81f0ad" src="https://user-images.githubusercontent.com/50660347/123896422-1c699200-d994-11eb-9b70-88cec56a085a.png">

![Uploading image.png…]()

**Comment by ttyio** [comment]
Hello @muzhibo1314 ,
Your plugin implementation is not correct, we need launch CUDA kernel in plugin `enqueue` function.
And for the reason why the size is so small, could you use netron to visualize your onnx model, check if the model has any input? thanks

**Comment by muzhibo1314** [comment]
> Hello @muzhibo1314 ,
> Your plugin implementation is not correct, we need launch CUDA kernel in plugin `enqueue` function.
> And for the reason why the size is so small, could you use netron to visualize your onnx model, check if the model has any input? thanks

Hello @ttyio ,
I use netron to visualize my onnx model ,this is the screenshot.
![1625728768281](https://user-images.githubusercontent.com/50660347/124879540-eb2a3b00-dfff-11eb-9578-44001e08bebc.jpg)
![1625728859225](https://user-images.githubusercontent.com/50660347/124879751-2167ba80-e000-11eb-88e3-656a8d062245.jpg)

I don’t know how to launch CUDA kernel in plugin `enqueue` function .I need to know what function information of the model about onehot. Could you give me some suggestions,Many thanks!

**Comment by muzhibo1314** [comment]
@ttyio Hello,According to my understanding, the `enqueue` function is not executed when the model is converted, but will only be executed during inference service. I don’t know if my understanding is correct. Many Thanks

**Comment by ttyio** [comment]
@muzhibo1314 , yes `enqueue` is not executed during parse. I just see this error so point it out.

And I want to check the network input and output, not the `onehot` input and outputs. 
Thanks

**Comment by muzhibo1314** [comment]
> @muzhibo1314 , yes `enqueue` is not executed during parse. I just see this error so point it out.
> 
> And I want to check the network input and output, not the `onehot` input and outputs.
> Thanks
@ttyio Hello，this is the network input and output  screenshot ，and I configured the corresponding model configuration file.
Thanks
![1625744565016](https://user-images.githubusercontent.com/50660347/124917334-466f2400-e026-11eb-82ad-bf1a80a5c750.jpg)
<img width="393" alt="WeChatd0cda8bbea359e026ee5cab21ba4f91e" src="https://user-images.githubusercontent.com/50660347/124917496-7e766700-e026-11eb-901c-8f4a3999b82c.png">

**Comment by ttyio** [comment]
Hello @muzhibo1314 , what's this configuration file used for? what tool are you using to generate the plan file? thanks

**Comment by muzhibo1314** [comment]
Hi @ttyio , This configuration file is a configuration file required by TIS.I use TensorRT  method to convert the .onnx to plan file.

**Comment by ttyio** [comment]
Thanks @muzhibo1314 , have you tried 8.0 using trtexec? If it still fails, could you share with us a repro for debugging? thanks!

**Comment by muzhibo1314** [comment]
Hi, @ttyio I tried the TensorRT 8.0 version, and the same as the original problem, the model was truncated, and the generated plan file was very small.How i share with you a repro.Many Thanks.

**Comment by ttyio** [comment]
@muzhibo1314 , could you upload to any cloud file sharing service like baidu pan, google drive? thanks

**Comment by muzhibo1314** [comment]
@ttyio  I sent the link of google drive to the email address on your homepage.Thanks

**Comment by ttyio** [comment]
Hello @muzhibo1314 , sorry for the delay response.

The `OneHotPlugin::getOutputDimensions` is incorrect, the whole network size is reduced because the problem size is wrongly reduced after this oneHot layer. could you check the the onnx spec in https://github.com/onnx/onnx/blob/master/docs/Operators.md#OneHot?

Thanks

**Comment by muzhibo1314** [comment]
Hello @ttyio, thanks for replying to my question.
I learned the getOutputDimensions used to specify the dimensions of output as a function of the input dimensions.
Your mean is I was wrongly set this oneHot layer output dimensions that caused the next layer not get the correct input dimensions?
Thanks

**Comment by ttyio** [comment]
@muzhibo1314 , now onehot is supported in 8.5, could you take a try? thanks!

**Comment by ttyio** [comment]
Closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!
